common/moduleconfig/moduleconfig.go

```diff
@@ -107,15 +117,99 @@ func setConfigDefaults(moduleDir string, config *ModuleConfig) error {
 			return fmt.Errorf("deploy files must be relative to the module directory")
 		}
 	}
-	for _, watch := range config.Watch {
-		if !isBeneath(moduleDir, watch) {
-			return fmt.Errorf("watch files must be relative to the module directory")
-		}
-	}
+
 	return nil
 }
 
 func isBeneath(moduleDir, path string) bool {
 	resolved := filepath.Clean(filepath.Join(moduleDir, path))
 	return strings.HasPrefix(resolved, strings.TrimSuffix(moduleDir, "/")+"/")
 }
+
+func replacementWatches(moduleDir, deployDir string) ([]string, error) {
+	goModPath := filepath.Join(moduleDir, "go.mod")
+	goModBytes, err := os.ReadFile(goModPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read %s: %w", goModPath, err)
+	}
+	goModFile, err := modfile.Parse(goModPath, goModBytes, nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse %s: %w", goModPath, err)
+	}
+
+	replacements := reflect.DeepCopy(goModFile.Replace)
```

Comment:
I think it would be cleaner to convert this into a `map[string]string` and pass that around rather than cloning and mutating this.

Reply:
Awesome, yeah. Will update. I saw this used in another part of the code like this so assumed there was a reason, but a map would simplify this. Thanks!

Reply:
In the other location we were updating a go.mod file with changes.

Reply:
ahhh 👍